documentation/tutorial/tutorial.actor.cpp

```diff
@@ -0,0 +1,470 @@
+/*
+ * fdbcli.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "flow/flow.h"
+#include "flow/Platform.h"
+#include "flow/DeterministicRandom.h"
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbclient/ReadYourWrites.h"
+#include <functional>
+#include <unordered_map>
+#include <memory>
+#include <iostream>
+#include "flow/actorcompiler.h"
+
+NetworkAddress serverAddress;
+
+// this is a simple actor that will report how long
+// it is already running once a second.
+ACTOR Future<Void> simpleTimer() {
+	// we need to remember the time when we first
+	// started.
+	// This needs to be a state-variable because
+	// we will use it in different parts of the
+	// actor. If you don't understand how state
+	// variables work, it is a good idea to remove
+	// the state keyword here and look at the
+	// generated C++ code from the actor compiler.
+	state double start_time = g_network->now();
+	loop {
+		wait(delay(1.0));
+		std::cout << format("Time: %.2f\n", g_network->now() - start_time);
+	}
+}
+
+// A actor that demonstrates how choose-when
+// blocks work.
+ACTOR Future<Void> someFuture(Future<int> ready) {
+	loop {
+		choose {
+			when(wait(delay(0.5))) { std::cout << "Still waiting...\n"; }
+			when(int r = wait(ready)) {
+				std::cout << format("Ready %d\n", r);
+				wait(delay(double(r)));
+				std::cout << "Done\n";
+				return Void();
+			}
+		}
+	}
+}
+
+ACTOR Future<Void> promiseDemo() {
+	state Promise<int> promise;
+	state Future<Void> f = someFuture(promise.getFuture());
+	wait(delay(3.0));
+	promise.send(2);
+	wait(f);
+	return Void();
+}
+
+ACTOR Future<Void> eventLoop(AsyncTrigger* trigger) {
+	loop {
```

Comment:
Actually there is another syntax like:

```
loop choose {
  when(wait(A)) {}
  when(wait(B)) {}
}
```

So I suggest write one of the examples in this style.

Reply:
I am not sure this needs another example as this is what one would expect (you don't need braces for a `while (true)` block and this is pretty much what it is). But I will add a comment